src/passes/OptimizeInstructions.cpp

```diff
@@ -1902,6 +1902,19 @@ struct OptimizeInstructions
     return HeapType::isSubType(a, b) || HeapType::isSubType(b, a);
   }
 
+  bool canBeCastTo(Type a, Type b) {
```

Comment:
Instead of having this function that returns `true` when the cast may succeed or will certainly succeed, how about we have just a single method that returns `Success`, `Failure`, or `Unknown` just by checking `a <: b` and `b <: a`? That code would be simpler and would give us more information.

Reply:
Makes sense. Would you mind if I leave that for a followup? I already have a PR for after this that would conflict.